### PR TITLE
feat(authenticator): Authenticator Connected Component

### PR DESCRIPTION
**Pull request by mattcreaser**
## Release PR review checklist
- [x] Verify VERSION_NAME in authenticator/gradle.properties
- [x] Verify authenticator/CHANGELOG.md

## Module versions

Authenticator 1.0.0-dev-preview.0

### CODE REVIEW

**Review by tjleing — APPROVED** [review]

File: authenticator/src/main/java/com/amplifyframework/ui/authenticator/forms/FieldConfig.kt [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    ) : FieldConfig()
+
+    /**
+     * Configuration object for a phone number input
+     * @param key The identifier key for this field.
+     * @param required Whether this field requires a non-blank value to submit the form.
+     * @param validator The [FieldValidator] instance for this field.
+     * @param label The label for this field. This is null for standard fields.
+     * @param hint The hint for this field. This is null for standard fields.
+     * @param defaultCountryCode The default country code that is pre-selected for this phone number.
```
Comment (eeatonaws) [inline authenticator/src/main/java/com/amplifyframework/ui/authenticator/forms/FieldConfig.kt:126]: Should this parameter be included in the constructor below?

Comment (mattcreaser) [reply authenticator/src/main/java/com/amplifyframework/ui/authenticator/forms/FieldConfig.kt:126]: Good catch, this param has been removed for now.

**Review by tjleing — DISMISSED** [review]

File: samples/authenticator/build.gradle [hunk 2]
```diff
@@ -0,0 +1,16 @@
+buildscript {
+    ext {
+        compose_version = '1.3.3'
+        amplifyVersion = '2.7.0'
```
Comment (eeatonaws) [inline samples/authenticator/build.gradle:4]: Should the Amplify version be 2.7.1?

Comment (mattcreaser) [reply samples/authenticator/build.gradle:4]: This can actually be removed.

**Review by eeatonaws — APPROVED** [review]

**Review by tjleing — APPROVED** [review]

**Pull request merged**